vault/resource_terraform_cloud_secret_creds.go

```diff
@@ -78,7 +80,11 @@ func readTerraformCloudSecretCredsResource(d *schema.ResourceData, meta interfac
 	organization := secret.Data["organization"]
 	teamId := secret.Data["team_id"]
 
-	d.SetId(tokenId)
+	if secret.LeaseID != "" {
+		d.SetId(secret.LeaseID)
+	} else {
+		d.SetId(tokenId)
+	}
```

Comment:
Would this mean that any reference to the lease where it's assumed that the value is the lease ID won't work? For instance `deleteTerraformCloudSecretCredsResource` assumes that `d.Id()` is a lease ID and will call revoke using this value, which doesn't work if `d.SetId(tokenId)`.

Reply:
The failure of the lease lookup for Org/Team tokens is fine as there is no dynamic cred to revoke/delete.

Reply:
Would it make sense to always set the ID to `tokenId`, and only `d.Set("lease_id", secret.LeasedID)` if that's non-empty. That way you can call lease revoke/lookup if the lease_id is non-empty. @catsby thoughts on this approach?

Reply:
I went through and added the `lease_id` attribute. Ran into an issue where `lease_id` needed to be set for every token. I moved it to an empty string if no lease is created.